File: frontend/src/api/caja.ts
import { api } from "./client";

export interface Caja {
  id: string;
  usuarioId: string;
  montoApertura: number;
  montoCierre: number | null;
  abiertaAt: string;
  cerradaAt: string | null;
  estado: string;
  observaciones: string | null;
}

export const cajaApi = {
  estado: () => api.get<{ actual: Caja | null; historial: Caja[]; ventasEnCaja: number }>("/api/caja").then(r => r.data),
  abrir: (montoApertura: number) => api.post<Caja>("/api/caja/abrir", { montoApertura }).then(r => r.data),
  cerrar: (id: string, montoCierre: number, observaciones: string) =>
    api.post<Caja>(`/api/caja/${id}/cerrar`, { montoCierre, observaciones }).then(r => r.data),
};

File: backend/src/main/java/com/farmacia/model/Caja.java
package com.farmacia.model;

import jakarta.persistence.*;
import lombok.Data;

import java.math.BigDecimal;
import java.time.OffsetDateTime;
import java.util.UUID;

@Data
@Entity
@Table(name = "cajas")
public class Caja {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(name = "usuario_id", nullable = false)
    private UUID usuarioId;

    @Column(name = "monto_apertura", nullable = false, precision = 12, scale = 2)
    private BigDecimal montoApertura = BigDecimal.ZERO;

    @Column(name = "monto_cierre", precision = 12, scale = 2)
    private BigDecimal montoCierre;

    @Column(name = "abierta_at", nullable = false)
    private OffsetDateTime abiertaAt;

    @Column(name = "cerrada_at")
    private OffsetDateTime cerradaAt;

    @Column(nullable = false)
    private String estado = "abierta";

    private String observaciones;
}
